### PR TITLE
Use dependency injection

### PR DESCRIPTION
**Pull request by el7cosmos**

### CODE REVIEW

**Comment by joaogarin** [comment]
Thanks for the PR: will try have a look at it as soon as possible

**Comment by duartegarin** [comment]
Hi guys,
Currently away but I will try to review the PR this week.
Thanks

File: src/Plugin/GraphQL/Derivers/SearchAPIDocumentTypeDeriver.php [hunk 1]
```diff
@@ -3,17 +3,32 @@
...
+  /**
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
 
   /**
    * Constructs a new Search API document type.
    */
-  public function __construct() {
+  public function __construct(EntityTypeManagerInterface $entity_type_manager) {
```
Comment (joaogarin) [inline src/Plugin/GraphQL/Derivers/SearchAPIDocumentTypeDeriver.php:26]: Here we should now replace the comment with the proper PHPDoc signature for the arguments in the constructor.
Something like : 
```
/**
 * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
 *   The entity type manager service.
 */
```

File: src/Plugin/GraphQL/Derivers/SearchAPIFieldDeriver.php [hunk 2]
```diff
@@ -3,18 +3,33 @@
...
+  /**
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
 
   /**
    * Constructs a new Search API field.
    */
-  public function __construct() {
+  public function __construct(EntityTypeManagerInterface $entity_type_manager) {
```
Comment (joaogarin) [inline src/Plugin/GraphQL/Derivers/SearchAPIFieldDeriver.php:27]: Same as above we can now include the PHPDoc signature here withe information on the params passed to the constructor.

**Review by joaogarin — CHANGES_REQUESTED** [review]
Thanks for the PR I gave it a test and it looks good! 👍 I think it improves the code quite a lot so we should merge this. just minor coding standards stuff that [coder](https://www.drupal.org/project/coder) is telling me about but nothing big. If you cant do these changes I can also include them np :) Thanks for the pull request

**Review by joaogarin — APPROVED** [review]
Looks good to me 👍

**Comment by duartegarin** [comment]
Thank you @el7cosmos

**Pull request merged**